pkgs/applications/networking/ouroboros/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv
+, lib
+, cmake
+, dnsutils
+, fetchgit
+, fuse
+, libgcrypt
+, openssl
+, pkg-config
+, protobuf
+, protobufc
+, systemd
+}:
+stdenv.mkDerivation rec {
+  pname = "ouroboros";
+  version = "0.20.1";
+  src = fetchgit {
+    url = "git://ouroboros.rocks/${pname}";
+    rev = version;
+    hash = "sha256-r5IyWEdNDfrJnW8Th+pWvGVFHsxEp91IQomesLNADJI=";
+  };
+  buildInputs = [
+    cmake
+    dnsutils
+    fuse
+    libgcrypt
+    openssl
+    pkg-config
+    protobuf
+    protobufc
+    systemd
+  ];
```

Comment:
Some of these should be in `nativeBuildInputs` instead, `cmake` and `pkg-config` at the very least seem like strong contenders for it. Please properly separate `buildInputs` from `nativeBuildInputs`.

To enforce proper separation in the derivation, you can set `strictDeps = true`. Aside from that, it will be necessary for i.e. `pkgsCross.aarch64-multiplatform.ouroboros` to work.

Reply:
I've separated what I can, but unfortunately the package itself doesn't make a good distinction here, requiring (the same?) `protoc-c` apparently both at build time and at runtime.